codec-http/src/main/java/io/netty/handler/codec/http/HttpContentCompressor.java

```diff
@@ -83,47 +95,68 @@ public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel)
+     *                             value should be in the range {@code 9} to {@code 15} inclusive.
+     *                             Larger values result in better compression at the expense of
+     *                             memory usage.  The default value is {@code 15}.
+     * @param memLevel             How much memory should be allocated for the internal compression
+     *                             state.  {@code 1} uses minimum memory and {@code 9} uses maximum
+     *                             memory.  Larger values result in better and faster compression
+     *                             at the expense of memory usage.  The default value is {@code 8}
+     * @param contentSizeThreshold The response body is compressed when the size of the response
+     *                             body exceeds the threshold. The value should be a non negative
+     *                             number. {@code 0} will enable compression for all responses.
      */
     public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel, int contentSizeThreshold) {
+        this(compressionLevel, windowBits, memLevel, contentSizeThreshold, 4);
+    }
+
+    /**
+     * Creates a new handler with the specified compression level, window size,
+     * and memory level..
+     *
+     * @param compressionLevel     {@code 1} yields the fastest compression and {@code 9} yields the
+     *                             best compression.  {@code 0} means no compression.  The default
+     *                             compression level is {@code 6}.
+     * @param windowBits           The base two logarithm of the size of the history buffer.  The
+     *                             value should be in the range {@code 9} to {@code 15} inclusive.
+     *                             Larger values result in better compression at the expense of
+     *                             memory usage.  The default value is {@code 15}.
+     * @param memLevel             How much memory should be allocated for the internal compression
+     *                             state.  {@code 1} uses minimum memory and {@code 9} uses maximum
+     *                             memory.  Larger values result in better and faster compression
+     *                             at the expense of memory usage.  The default value is {@code 8}
+     * @param contentSizeThreshold The response body is compressed when the size of the response
+     *                             body exceeds the threshold. The value should be a non negative
+     *                             number. {@code 0} will enable compression for all responses.
+     * @param brotliCompressionQuality Brotli Compression Quality. The default compression quality is {@code 4}.
+     */
+    public HttpContentCompressor(int compressionLevel, int windowBits, int memLevel, int contentSizeThreshold,
+                                 int brotliCompressionQuality) {
```

Comment:
Instead of adding more and more constructors, we should come up with an API that scales to more compression algorithms. We have at least ZSTD that we'd also like to see added. I'm thinking we could have a `ContentEncoding` enum, and have the compression levels passed as an `EnumMap` from `ContentEncoding` to `Integer`.

For reference, the full list in the content encoding registry is here: https://www.iana.org/assignments/http-parameters/http-parameters.xhtml